### PR TITLE
Integrate Damian's zlib tests

### PR DESCRIPTION
**Pull request by AeroStun**
Step n°1 for #1719

Damian:
- Added an example
- Fixed data races in http and websocket examples
- Switched Azure Pipelines to Clang 9
- Added documentation for configuration macros
- Added various tests for zlib

Builds and all tests pass on my machine; will see what the CI says
(Had to use my own branch so I could cherry-pick and skip one commit that clashed with upstream)

### CODE REVIEW

**Comment by vinniefalco** [comment]
There are commits unrelated to zlib, these need to go in their own pull request

**Comment by AeroStun** [comment]
Okay; I'll redispatch them

**Comment by vinniefalco** [comment]
Where are the other commits?

**Comment by AeroStun** [comment]
Nowhere for now; I've been focusing on fixing these ones

**Comment by vinniefalco** [comment]
The other commits have to go in first. If you roll them up into a pull request then I will review it and merge it.

**Comment by AeroStun** [comment]
Okay; will do

**Comment by AeroStun** [comment]
Since Travis doesn't seem to start the build here, I triggered it on my fork, and everything passes
https://travis-ci.org/AeroStun/beast/builds/594341155

File: test/beast/zlib/inflate_stream.cpp [hunk 1]
```diff
@@ -360,6 +380,143 @@ class inflate_stream_test : public beast::unit_test::suite
...
+        check({0x05,0xe0, 0x81, 0x91, 0x24, 0xcb, 0xb2, 0x2c,
+               0x49, 0xe2, 0x0f, 0x2e, 0x8b, 0x9a, 0x47, 0x56,
+               0x9f, 0xfb, 0xfe, 0xec, 0xd2, 0xff, 0x1f},
+            error::end_of_stream);
+        check({0xed, 0xc0, 0x01, 0x01, 0x00, 0x00, 0x00, 0x40,
+               0x20, 0xff, 0x57, 0x1b, 0x42, 0x2c, 0x4f},
+            error::end_of_stream);
+        check({0x02, 0x08, 0x20, 0x80, 0x00, 0x03, 0x00},
+            error::end_of_stream);
+        // TODO: Excess data (from golang test inflate suite), should this be an error?
```
Comment (AeroStun) [inline test/beast/zlib/inflate_stream.cpp]: IMO excess data in inflate is definitely an error; the user should be able to provide the right compressed length

**Comment by vinniefalco** [comment]
Does it fail on regular zlib?

**Comment by AeroStun** [comment]
Looks like it yes: https://github.com/madler/zlib/blob/cacf7f1d4e3d44d871b605da3b647f07d718623f/inflate.c#L891

**Comment by AeroStun** [comment]
So I guess we'll simply pop off the comment
Any changes you'd want to be done with it ?

**Comment by vinniefalco** [comment]
Not sure, I have to review it and things are getting heated on the wg21 mailing lists...

**Pull request closed without merging**